torch/onnx/_internal/exporter.py

```diff
@@ -522,12 +451,38 @@ def export(self) -> ExportOutput:
             *self.model_args, **self.model_kwargs
         )
 
-        # Export FX graph to ONNX ModelProto.
-        #
-        # Note that ALL kwargs are folded into constants in graph_module, so we don't pass kwargs
-        # to _export.
-        return self.options.fx_tracer._export_fx_to_onnx(
-            self.options, graph_module, updated_model_args
+        # TODO: Design the passes API
+        graph_module = pre_export_passes(self.options, graph_module, updated_model_args)
+
+        from onnxscript.function_libs.torch_lib import (  # type: ignore[import]
+            graph_building as onnxscript_graph_building,
+        )
+
+        onnxscript_graph = onnxscript_graph_building.TorchScriptGraph()
+
+        from torch.onnx._internal.fx import fx_onnx_interpreter
+
+        fx_interpreter = fx_onnx_interpreter.FxOnnxInterpreter(
```

Comment:
Move
```
onnxscript_graph_building.TorchScriptTracingEvaluator(...)
```
out from function call. E.g.,
```
evaluator = onnxscript_graph_building.TorchScriptTracingEvaluator(...)

fx_interpreter.run(  torchscript_tracer=onnxscript_graph_building.TorchScriptTracingEvaluator(evaluator, evaluator.graph, ...)
```

Reply:
Do we need access to the evaluator outside of `fx_interpreter`?

Reply:
I was lost for a moment then realized this `evaluator` was called `tracer` in old code lol, and there is another `evaluator` which is the `onnxscript.evaluator`.

We don't. So I'm also leaning towards not lifting the scope beyond `fx_interpreter`. https://github.com/pytorch/pytorch/pull/102810#discussion_r1231567639

Reply:
> Do we need access to the evaluator outside of `fx_interpreter`?

no, moved to inside `run`